### ISSUE TITLE
GCC compiler warnings with lto activated

### ISSUE THREAD

**Issue by mgerhardy** [ufbx/ufbx#70]
```
 /home/runner/work/vengi/vengi/src/modules/voxelformat/external/ufbx.h:4204:35: warning: type of ‘ufbx_axes_right_handed_y_up’ does not match original declaration [-Wlto-type-mismatch]
 4204 | extern const ufbx_coordinate_axes ufbx_axes_right_handed_y_up;
      |                                   ^
/home/runner/work/vengi/vengi/src/modules/voxelformat/external/ufbx.c:25036:28: note: ‘ufbx_axes_right_handed_y_up’ was previously declared here
25036 | const ufbx_coordinate_axes ufbx_axes_right_handed_y_up = {
      |                            ^
```

**Comment by bqqbarbhg** [comment]
Heya! Are you including the header in C++ by chance? I have to enforce `enum` types to be 32 bits wide and to make the library nicer for users in C++ it uses a `enum name : uint32_t` declaration, but in pre-C23 C it has to add a dummy enum value to force the width to 32 bits.. As `ufbx_coordinate_axes` consists of these enums I think the compiler treats these two definitions as distinct.

As all fixes to this have some annoying tradeoffs I'll need to think about how to fix this for a bit.

**Comment by bqqbarbhg** [comment]
Fixed this in the end in (#75 v0.6.0) by having explicit `enum` width disabled by default forcing us to have the `_FORCE_32BIT` enum values on C++ and C11..
